src/app/stack/card-details/card-details.component.ts

```diff
@@ -981,9 +1320,62 @@ export class CardDetailsComponent implements OnInit, OnChanges {
         return headers;
     }
 
+    private calBadge(report: MReportInformation, name: string): string {
+
+
+        let privateVulnerabilitiesSet = new Set();
+        let publicVulnerabilitiesSet = new Set();
+
+        report.componentDetails.forEach(element => {
+
+            if (element.componentInformation.public_vulnerabilities && element.componentInformation.public_vulnerabilities.length > 0) {
+                element.componentInformation.public_vulnerabilities.forEach(pubDep => {
+                    publicVulnerabilitiesSet.add(pubDep)
+                });
+            }
+            if (element.componentInformation.private_vulnerabilities && element.componentInformation.private_vulnerabilities.length > 0) {
+                element.componentInformation.private_vulnerabilities.forEach(priDep => {
+                    privateVulnerabilitiesSet.add(priDep)
+                });
+            }
+
+            if (element.componentInformation.vulnerable_dependencies && element.componentInformation.vulnerable_dependencies.length > 0) {
+
+                element.componentInformation.vulnerable_dependencies.forEach(tdep => {
+                    if (tdep.public_vulnerabilities && tdep.public_vulnerabilities.length > 0) {
+                        tdep.public_vulnerabilities.forEach(transPubDep => {
+                            publicVulnerabilitiesSet.add(transPubDep)
+                        });
+                    }
+                    if (tdep.private_vulnerabilities && tdep.private_vulnerabilities.length > 0) {
+                        tdep.private_vulnerabilities.forEach(transPriDep => {
+                            privateVulnerabilitiesSet.add(transPriDep)
+                        });
+                    }
+                });
+
+            }
+        });
+
+        if (publicVulnerabilitiesSet.size || privateVulnerabilitiesSet.size) {
+            if (name == "Commonly Known Vulnerabilities") {
+                return publicVulnerabilitiesSet.size + '';
+            }
+            if (name == "Vulnerabilities unique to Snyk") {
+                return privateVulnerabilitiesSet.size + '';
+            }
+        } else {
+            return '0'
+        }
```

Comment:
Use `toString` please. BTW what do you really need to convert to string explicitly ? It would be automatically converted when using along with string.

Reply:
Yes, it will be converted to string.
It's just that the function has a return type of string. Just to keep it consistent everywhere.

I have updated it to toString().